### PR TITLE
add assertions about metrics

### PR DESCRIPTION
**Pull request by richardstartin**
Recovers the metrics asserts from the abandoned #1484

Also interns some constant metric names I spotted.

### CODE REVIEW

File: dd-trace-core/src/test/groovy/datadog/trace/core/StringTablesTest.groovy [hunk 1]
```diff
@@ -23,8 +23,8 @@ class StringTablesTest extends Specification {
       approxSizeBytesIgnoringHashMapOverhead += entry.getKey().length()
       approxSizeBytesIgnoringHashMapOverhead += entry.getValue().length
     }
-    // less than 3KB memory overhead seems reasonable for reducing GBs of allocations
-    expect: approxSizeBytesIgnoringHashMapOverhead < 1024 * 3
+    // less than 4KB memory overhead seems reasonable for reducing GBs of allocations
```
Comment (richardstartin) [inline dd-trace-core/src/test/groovy/datadog/trace/core/StringTablesTest.groovy:26]: let's really revisit this if it gets above 4KB (breached 3KB by about 10 bytes)

Comment (richardstartin) [inline dd-trace-core/src/test/groovy/datadog/trace/core/StringTablesTest.groovy:26]: let's really revisit this at 4KB, this breached 3KB by about 10 bytes and I think we have them all now.

**Comment by richardstartin** [comment]
I don't know why the finatra server test is failing, but it does so consistently on test_latest8, and these changes should not provoke this change in behaviour:

```
java.lang.ExceptionInInitializerError
	at com.twitter.finagle.stats.JsonExporter.<init>(JsonExporter.scala:78)
	at com.twitter.finagle.stats.JsonExporter.<init>(JsonExporter.scala:82)
	at com.twitter.finagle.stats.MetricsExporter.<init>(MetricsStatsReceiver.scala:160)
	at com.twitter.finagle.stats.MetricsExporter.<init>(MetricsStatsReceiver.scala:167)
	at com.twitter.finagle.stats.MetricsExporter.<init>(MetricsStatsReceiver.scala:169)
	at java.lang.Class.newInstance(Class.java:442)
	at com.twitter.app.LoadService$$anonfun$loadImpls$2.apply(LoadService.scala:192)
	at com.twitter.app.LoadService$$anonfun$loadImpls$2.apply(LoadService.scala:181)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at com.twitter.app.LoadService$.loadImpls(LoadService.scala:181)
	at com.twitter.app.LoadService$.apply(LoadService.scala:135)
	at com.twitter.app.LoadService$.apply(LoadService.scala:149)
	at com.twitter.finagle.util.LoadService$.apply(LoadService.scala:14)
	at com.twitter.finagle.http.HttpMuxer$.<init>(HttpMuxer.scala:177)
	at com.twitter.server.Lifecycle$class.$init$(Lifecycle.scala:14)
	at FinatraServer.<init>(FinatraServer.scala:6)
	at FinatraServerTest.startServer(FinatraServerTest.groovy:30)
	at datadog.trace.agent.test.base.HttpServerTest.setupSpec(HttpServerTest.groovy:66)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.11.0 requires Jackson Databind version >= 2.11.0 and < 2.12.0
	at com.fasterxml.jackson.module.scala.JacksonModule$class.setupModule(JacksonModule.scala:61)
	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:751)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModules(ObjectMapper.java:932)
	at com.twitter.util.tunable.JsonTunableMapper.<init>(JsonTunableMapper.scala:117)
	at com.twitter.util.tunable.JsonTunableMapper$.apply(JsonTunableMapper.scala:39)
	at com.twitter.util.tunable.JsonTunableMapper$.apply(JsonTunableMapper.scala:45)
	at com.twitter.finagle.tunable.StandardTunableMap$$anonfun$loadJsonConfig$1.apply(StandardTunableMap.scala:112)
	at com.twitter.finagle.tunable.StandardTunableMap$$anonfun$loadJsonConfig$1.apply(StandardTunableMap.scala:110)
	at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
	at scala.collection.immutable.List.foldLeft(List.scala:84)
	at com.twitter.finagle.tunable.StandardTunableMap$.loadJsonConfig(StandardTunableMap.scala:110)
	at com.twitter.finagle.tunable.StandardTunableMap$$anonfun$1.apply(StandardTunableMap.scala:39)
	at com.twitter.finagle.tunable.StandardTunableMap$$anonfun$1.apply(StandardTunableMap.scala:38)
	at com.twitter.finagle.tunable.StandardTunableMap$$anon$1.apply(StandardTunableMap.scala:55)
	at com.twitter.finagle.tunable.StandardTunableMap$$anon$1.apply(StandardTunableMap.scala:54)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at com.twitter.finagle.tunable.StandardTunableMap$.apply(StandardTunableMap.scala:52)
	at com.twitter.finagle.tunable.StandardTunableMap$.apply(StandardTunableMap.scala:48)
	at com.twitter.finagle.stats.package$.<init>(package.scala:11)
	... 25 more
```

**Comment by tylerbenson** [comment]
Does that test fail locally?  Does it also fail on master?

**Comment by devinsba** [comment]
For the test failure: https://github.com/open-telemetry/opentelemetry-auto-instr-java/pull/450

**Pull request merged**